src/components/loader-spinner/loader-spinner.spec.tsx

```diff
@@ -56,6 +56,20 @@ describe("LoaderSpinner", () => {
   });
 
   describe("when custom props are passed", () => {
+    it("when the 'spinnerLabel' prop is passed a custom string value it overrides the default visible label", () => {
+      render(<LoaderSpinner spinnerLabel="foo" />);
+      const visibleLabelElement = screen.getByTestId("visible-label");
+
+      expect(visibleLabelElement).toHaveTextContent("foo");
+    });
+
+    it("when the 'spinnerLabel' prop is passed a custom string value it overrides the visually hidden label", () => {
+      render(<LoaderSpinner spinnerLabel="bar" showSpinnerLabel={false} />);
+      const hiddenLabelElement = screen.getByTestId("hidden-label");
+
+      expect(hiddenLabelElement).toHaveTextContent("bar");
```

Comment:
**suggestion(non-blocking)**: what do you think about checking the label is visually hidden on the page instead of using test ids? That way, it would mimick how an AT user would receive the "bar" message:  

```suggestion
      // import { within } from "@testing-library/react";

      const liveRegion = screen.getByRole("status");
      const hiddenLabel = within(hiddenLabel).getByText("bar");

      expect(hiddenLabel).toHaveStyle({
        border: "0",
        height: "1px",
        margin: "-1px",
        overflow: "hidden",
        padding: "0",
        position: "absolute",
        width: "1px",
        "white-space": "nowrap",
      });
```

Reply:
I don't mind this approach either, but won't this test become fairly brittle as it will fail every time any of these CSS attributes are slightly changed going forward?

Reply:
Good point 👍🏼 Maybe the `toBeInTheDocument` assertion would a better fit in that case?

Reply:
I think it probably would, but we still have the dilemma that `getByText` will just assert that that specific text is found within the live region, and does not indicate whether the text found is visually hidden or not. Whereas the test-id approach does, as it directly points to the specific DOM element